src/main/java/org/spongepowered/api/entity/living/Human.java

```diff
@@ -119,6 +146,79 @@
     void setTotalExperience(double totalExperience);
 
     /**
+     * Returns whether this human entity is sneaking.
+     *
+     * @return Whether this human entity is sneaking
+     */
+    boolean isSneaking();
+
+    /**
+     * Sets whether this human entity is sneaking.
+     *
+     * @param sneaking Whether this human entity should be sneaking.
+     */
+    void setSneaking(boolean sneaking);
+
+    /**
+     * Gets whether this human entity is currently flying.
+     *
+     * @return Whether this human entity is currently flying
+     */
+    boolean isFlying();
+
+    /**
+     * Sets whether this human entity is currently flying.
+     *
+     * <p>Note: this will have no effect if {@link Human#isFlightAllowed()}
+     * returns <code>false</code>.</p>
+     *
+     * @param flying Whether this human entity should be flying
+     */
+    void setFlying(boolean flying);
+
+    /**
+     * Returns whether this human entity is allowed to fly.
+     *
+     * @return Whether this human entity is allowed to fly
+     */
+    boolean isFlightAllowed();
+
+    /**
+     * Sets if this human entity is allowed to fly.
+     *
+     * @param allowFlight Whether this human entity is allowed to fly.
+     */
+    void setFlightAllowed(boolean allowFlight);
+
+    /**
+     * Gets the speed this human entity may fly at.
+     *
+     * @return the speed this human entity may fly at
+     */
+    double getFlySpeed();
+
+    /**
+     * Sets the speed this human entity may fly at.
```

Comment:
You're missing some integral restrictions on the movement speed in general. Such as negative speeds severely affecting clients (or preventing them from even being able to interact with the server).